docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,220 @@
+:_content-type: PROCEDURE
+:myfunctiontwo: prefetch_hermetic_builds
+
+[id="prefetching-dependencies-to-support-hermetic-build_{context}"]
+= Prefetching the package manager dependencies for the Hermetic build
+
+In {ProductName}, hermetic builds are achieved by restrciting network access to the build. Consequently, builds using supported languages may fail if they cannot retrieve dependencies from respositories. To address this, {ProductName} uses Cachi2 to prefetch the package manager dependencies for the supported-languages. Cachi2 makes it easier for package managers to prefetch dependencies. It also helps generate a manifest of all dependencies included in your builds, ensuring transparency and maintainability.
+
+[#supported-languages]
+.Supported languages
+
+[cols="1,1"]
+|===
+|Language
+|Package Manager
+
+|Go
+|`gomod`
+
+|Python
+|`pip`
+|===
+
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have an link:https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/configuring-builds/proc_upgrade_build_pipeline/[upgraded build pipeline].
+* You have a `go.mod` file in your repository that lists all the the dependencies.
+
+.Procedures
+To create a hermetic build for a component, complete the following steps:
+
+. Go to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. Configure the hermetic pipeline by adding the following hermetic pipeline parameters in both the *.yaml* files:
```

Comment:
@arewm @chmeliik @gtrivedi88 Sorry, I don't think we brought this discussion back here, but from the discussion in a chat https://redhat-internal.slack.com/archives/C04GVLR0155/p1686920028643609 having them set spec.params is sufficient, but our default yaml we provide them includes in-lined values, which would be overridden if they add the same name to spec.params. If users don't understand that hierarchy, they might get confused and change more than necessary, or find and replace those existing values. 

You could consider a note in the procedures where they're adding these params that they can ignore the related pipelinespec default value.